addons/website_sale/static/src/js/product_configurator_dialog/product_configurator_dialog.xml

```diff
@@ -25,4 +25,13 @@
             </t>
         </xpath>
     </t>
+
+    <t t-inherit="sale.ProductList" t-inherit-mode="extension">
```

Comment:
Since the inherited template is defined in its own file, could you do the same here? (i.e. create a new `product_list.xml` file in a new `product_list` folder directly under the `js` folder). Sorry I didn't notice before.